internal/app/agent/service.go

```diff
@@ -168,41 +176,70 @@ func (a *agent) Control(uuid, cmdStr string) error {
 		return err
 	}
 
-	payload, err := encodeSenML(uuid, cmd, resp)
-	if err != nil {
-		return err
-	}
-
-	if err := a.Publish(a.config.Agent.Channels.Control, string(payload)); err != nil {
-		return err
-	}
-
-	return nil
+	return a.processResponse(resp, uuid, cmd)
 }
 
 // Message for this command
-// "[{"bn":"1:", "n":"config", "vs":"export, /configs/export/config.toml, config_file_content"}]"
+// [{"bn":"1:", "n":"services", "vs":"view"}]
+// [{"bn":"1:", "n":"config", "vs":"save, export, filename, filecontent"}]
 // config_file_content is base64 encoded marshaled structure representing service conf
 // Example of creation:
 // 	b, _ := toml.Marshal(cfg)
 // 	config_file_content := base64.StdEncoding.EncodeToString(b)
-func (a *agent) ServiceConfig(uuid, cmdStr string) error {
+func (a *agent) ServiceConfig(uuid, cmdStr string) (err error) {
 	cmdArgs := strings.Split(strings.Replace(cmdStr, " ", "", -1), ",")
-	if len(cmdArgs) < 3 {
+	if len(cmdArgs) < 1 {
 		return errInvalidCommand
 	}
+	services := []byte{}
+	resp := ""
+	cmd := cmdType(cmdArgs[0])
 
-	service := cmdArgs[0]
-	fileName := cmdArgs[1]
-	fileCont, err := base64.StdEncoding.DecodeString(cmdArgs[2])
+	switch cmd {
+	case view:
+		services, err = json.Marshal(a.Services())
```

Comment:
err ignored?

Reply:
it is not ignored it is returned at line 212

Reply:
But in line 201 you are using `services`

Reply:
line 212, but nevermind, I'll change it

Reply:
Line 201 will be executed before and `resp = string(services)` will fail if error is not null.